internal/plugin/connectors/tcp/mssql/connection_details.go

```diff
@@ -46,12 +54,15 @@ func NewConnectionDetails(credentials map[string][]byte) *ConnectionDetails {
 	switch {
 	case bytes.Equal(sslMode, []byte("disable")):
 		connDetails.SSLMode = string(credentials["sslmode"])
+		break
```

Comment:
Since this logic path will just go directly to the return on the bottom,I think this can be simplified a lot. Some pseudocode:
```golang
...
sslMode := string(credentials["sslmode"])
if sslMode == "disable" || sslMode == "" {
    connDetails.SSLMode = defaultSSLMode
    return connDetails
}

connDetails.parseSSLOptions(credentials)

return connDetails
```